DBus interface revamp

whot opened the pull request:
This is on top of #241 and should only be merged when that one is merged.

When I started tidying up the DBusInterface.md document I noticed a fair few inconsistencies with the API. A few basically superfluous calls and some imbalanced property handling (having a 'foo' readonly property but a SetFoo() method - instead we can just make the property read-write).

Note that his is not complete but I'm about to head out for GUADEC, so better to get this in now. The bits missing are changing the Button interface from using strings to using the libratbag enums and the LED type string to an enum too.

ratbagd.py will need to be synced to piper after merging this but no client changes are needed, this is all handled within the bindings.

Hjdskes commented on ratbagd/ratbagd-resolution.c:
```diff
@@ -250,16 +298,19 @@ const sd_bus_vtable ratbagd_resolution_vtable[] = {
-	SD_BUS_METHOD("SetReportRate", "u", "u", ratbagd_resolution_set_report_rate, SD_BUS_VTABLE_UNPRIVILEGED),
-	SD_BUS_METHOD("SetResolution", "uu", "u", ratbagd_resolution_set_resolution, SD_BUS_VTABLE_UNPRIVILEGED),
 	SD_BUS_METHOD("SetDefault", "", "u", ratbagd_resolution_set_default, SD_BUS_VTABLE_UNPRIVILEGED),
-	SD_BUS_SIGNAL("ActiveResolutionChanged", "u", 0),
```
Again: you forgot to remove this signal and the one below it from the bindings.

Hjdskes commented on ratbagd/ratbagd.c:
```diff
@@ -350,10 +349,8 @@ static inline void ratbagd_init_test_device(struct ratbagd *ctx) {}
-	SD_BUS_PROPERTY("Devices", "ao", ratbagd_get_devices, 0, 0),
+	SD_BUS_PROPERTY("Devices", "ao", ratbagd_get_devices, 0, SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
 	SD_BUS_PROPERTY("Themes", "as", ratbagd_get_themes, 0, 0),
-	SD_BUS_SIGNAL("DeviceNew", "o", 0),
```
Again: you forgot to remove this signal and the one below it from the bindings

Hjdskes commented:
(Grr.. commenting on individual commits makes them show up as outdated)

whot commented on tools/ratbagd.py:
```diff
@@ -151,7 +152,30 @@ def _get_dbus_property(self, property):
+        # Take our real value and wrap it into a variant. To call
+        # org.freedesktop.DBus.Properties.Set we need to wrap that again
+        # into a (ssv), where v is our value's variant.
+        # args to .Set are "interface name", "function name",  value-variant
```
it took me long enough to figure out how to send this, so I'd rather have the comment there to clarify what the code does ;)

whot commented on tools/ratbagd.py:
```diff
@@ -111,7 +111,11 @@ def __init__(self, interface, object_path):
+            except GLib.Error:
+                raise RatbagdDBusUnavailable()
+
             if _RatbagdDBus._dbus is None:
```
fwiw, the Is None thing was part of the other PR and is already merged, so I'll keep this local and push it with some other PR

Hjdskes replied:
I know it's not part of this PR; I was trying to say that according to the documentation checking for `None` **and** catching the exception isn't necessary.

whot commented:
Does it matter that we change the interface? ratbagd had a note from day one that this interface will change and we're not finished with the changes. the version number is also a bit of a hint :)

Anyway, addressed all o @Hjdskes comments and squashed them in, thanks for the reviews

bentiss commented:
Then we are good to merge :)
